dance/datasets/singlemodality.py

```diff
@@ -280,41 +280,41 @@ def _raw_to_dance(self, raw_data: Tuple[ad.AnnData, np.ndarray]):
         return data
 
 
-@dataclass
-class ImputationDatasetParams:
-    data_dir = None
-    random_seed = None
-    min_counts = None
-    train_dataset = None
-    test_dataset = None
-    gpu = None
-    filetype = None
+@register_dataset("imputation")
+class ImputationDataset(BaseDataset):
 
+    URL = {
+        "pbmc_data": "https://www.dropbox.com/s/brj3orsjbhnhawa/5k.zip?dl=0",
+        "mouse_embryo_data": "https://www.dropbox.com/s/8ftx1bydoy7kn6p/GSE65525.zip?dl=0",
+        "mouse_brain_data": "https://www.dropbox.com/s/zzpotaayy2i29hk/neuron_10k.zip?dl=0",
+        "human_stemcell_data": "https://www.dropbox.com/s/g2qua2j3rqcngn6/GSE75748.zip?dl=0"
+    }
 
-class ImputationDataset():
+    DATASET_TO_FILE = {
+        "pbmc_data": "5k_pbmc_protein_v3_filtered_feature_bc_matrix.h5",
+        "mouse_embryo_data": [
+            osp.join("GSE65525", i)
+            for i in [
+                "GSM1599494_ES_d0_main.csv",
+                "GSM1599497_ES_d2_LIFminus.csv",
+                "GSM1599498_ES_d4_LIFminus.csv",
+                "GSM1599499_ES_d7_LIFminus.csv",
+            ]
+        ],
+        "mouse_brain_data": "neuron_10k_v3_filtered_feature_bc_matrix.h5",
+        "human_stemcell_data": "GSE75748/GSE75748_sc_time_course_ec.csv.gz"
+    }  # yapf: disable
 
-    def __init__(self, random_seed=10, gpu=-1, filetype=None, data_dir="data", train_dataset="human_stemcell",
-                 test_dataset="pbmc", min_counts=1):
-        self.params = ImputationDatasetParams
-        self.params.data_dir = data_dir
-        self.params.random_seed = random_seed
-        self.params.min_counts = min_counts
-        self.params.train_dataset = train_dataset
-        self.params.test_dataset = test_dataset
-        self.params.gpu = gpu
-        self.params.filetype = filetype
+    def __init__(self, data_dir="data", dataset="human_stemcell", train_size=0.1):
+        super().__init__(data_dir, full_download=False)
+        self.data_dir = data_dir
+        self.dataset = dataset
+        self.train_size = train_size
```

Comment:
Similar to the previous comment, is `train_size` used? Let's find out a list of unused attrs in the updated `ImputationDataset` and remove them for clarity.